presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -335,7 +335,18 @@ private PartitionStatistics updatePartitionStatistics(PartitionStatistics oldPar
                 firstPresent(newBasicStatistics.getRowCount(), oldBasicStatistics.getRowCount()),
                 firstPresent(newBasicStatistics.getInMemoryDataSizeInBytes(), oldBasicStatistics.getInMemoryDataSizeInBytes()),
                 firstPresent(newBasicStatistics.getOnDiskDataSizeInBytes(), oldBasicStatistics.getOnDiskDataSizeInBytes()));
-        return new PartitionStatistics(updatedBasicStatistics, newPartitionStats.getColumnStatistics());
+        Map<String, HiveColumnStatistics> updatedColumnStatistics =
+                updateColumnStatistics(oldPartitionStats.getColumnStatistics(), newPartitionStats.getColumnStatistics());
+        return new PartitionStatistics(updatedBasicStatistics, updatedColumnStatistics);
+    }
+
+    private Map<String, HiveColumnStatistics> updateColumnStatistics(Map<String, HiveColumnStatistics> oldColumnStats, Map<String, HiveColumnStatistics> newColumnStats)
+    {
+        Map<String, HiveColumnStatistics> result = new HashMap<>(newColumnStats);
+        result.putAll(oldColumnStats.entrySet().stream()
+                .filter(entry -> !newColumnStats.containsKey(entry.getKey()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
```

Comment:
What's the benefit of this compared with
```
Map<String, HiveColumnStatistics> result = new HashMap<>(oldColumnStats);
result.putAll(newColumnStats);
```
?

Reply:
Sorry, misunderstood your comment.